tests/sklearn/test_sklearn_autolog.py

```diff
@@ -1462,3 +1462,257 @@ def test_metric_computation_handles_absent_labels():
+        mean_squared_error_data1 = sklmetrics.mean_squared_error(eval1_y, pred1_y)
+        r2_score_data2 = sklmetrics.r2_score(eval2_y, pred2_y)
+        lor_score_data1 = model.score(eval1_X, eval1_y)
+        mean_squared_error_data2 = sklmetrics.mean_squared_error(eval2_y, pred2_y, squared=False)
+
+        scorer1 = sklmetrics.make_scorer(sklmetrics.mean_squared_error, squared=False)
+        mean_squared_error3_data2 = scorer1(model, eval2_X, eval2_y)
+
+        eval1_X, eval1_y = eval1_X.copy(), eval1_y.copy()
+        # In metric key, it will include dataset name as "eval1_X-2"
+        lor_score_data1_2 = model.score(eval1_X, eval1_y)
+
+        # In metric key, it will include dataset name as "unknown_dataset"
+        lor_score_data1_3 = model.score(eval1_X.copy(), eval1_y.copy())
+
+        metric_info = load_json_artifact("metric_info.json")
+
+    run_id = run.info.run_id
+    params, metrics, tags, artifacts = get_run_data(run_id)
+    post_training_metrics = {k: v for k, v in metrics.items() if not k.startswith('training_')}
+
+    assert post_training_metrics == {
+        'r2_score_eval1_X': r2_score_data1,
+        'mean_squared_error_eval1_X': mean_squared_error_data1,
+        'r2_score-2_eval2_X': r2_score_data2,
+        'LogisticRegression_score_eval1_X': lor_score_data1,
+        'mean_squared_error-2_eval2_X': mean_squared_error_data2,
+        'mean_squared_error-3_eval2_X': mean_squared_error3_data2,
+        'LogisticRegression_score-2_eval1_X-2': lor_score_data1_2,
+        'LogisticRegression_score-3_unknown_dataset': lor_score_data1_3,
+    }
+
+    assert metric_info == {
+        'LogisticRegression_score-2_eval1_X-2': 'LogisticRegression.score(X=eval1_X, y=eval1_y)',
+        'LogisticRegression_score-3_unknown_dataset': 'LogisticRegression.score(X=<ndarray>, y=<ndarray>)',
+        'LogisticRegression_score_eval1_X': 'LogisticRegression.score(X=eval1_X, y=eval1_y)',
+        'mean_squared_error-2_eval2_X': 'mean_squared_error(y_true=eval2_y, y_pred=pred2_y, squared=False)',
+        'mean_squared_error-3_eval2_X': 'mean_squared_error(y_true=eval2_y, y_pred=y_pred, squared=False)',
+        'mean_squared_error_eval1_X': 'mean_squared_error(y_true=eval1_y, y_pred=pred1_y)',
+        'r2_score-2_eval2_X': 'r2_score(y_true=eval2_y, y_pred=pred2_y)',
+        'r2_score_eval1_X': 'r2_score(y_true=eval1_y, y_pred=pred1_y)'
+    }
+
+    mlflow.sklearn.autolog(disable=True)
+
+    # Test patched methods generate the same results with unpatched methods.
+    mean_squared_error_data1_original = sklmetrics.mean_squared_error(eval1_y, pred1_y)
+    assert np.isclose(mean_squared_error_data1_original, mean_squared_error_data1)
+
+    lor_score_data1_original = model.score(eval1_X, eval1_y)
+    assert np.isclose(lor_score_data1_original, lor_score_data1)
+
+    pred1_y_original = model.predict(eval1_X)
+    assert np.allclose(pred1_y_original, pred1_y)
+
+
+def test_run_all_metric_examples():
```

Comment:
This is a super cool test! Nice!

Reply:
Thanks for including [my suggestion](https://github.com/mlflow/mlflow/pull/4491#issuecomment-874433278)!

Reply:
Can we parametrize this test to make it easier to debug?